frontend/rust-lib/flowy-date/src/event_handler.rs

```diff
@@ -0,0 +1,38 @@
+use chrono::{Datelike, NaiveDate};
+use date_time_parser::DateParser;
+use fancy_regex::Regex;
+use flowy_error::FlowyError;
+use lib_dispatch::prelude::{data_result_ok, AFPluginData, DataResult};
+
+use crate::entities::*;
+
+#[tracing::instrument(level = "debug", skip_all, err)]
+pub(crate) async fn query_date_handler(
+  data: AFPluginData<DateQueryPB>,
+) -> DataResult<DateResultPB, FlowyError> {
+  let query: String = data.into_inner().query;
+  let date = DateParser::parse(&query);
+
+  let formatted: String;
+  if date.is_some() {
+    let naive_date = date.unwrap();
```

Comment:
It seems like you are using lot of unwrap. Please don't use unwrap unless it's safe to unwrap